app/controllers/rentals_controller.rb

```diff
@@ -23,20 +34,42 @@ def index
   # GET /rentals/1
   def show; end
 
-  # GET /rentals/cost?end_time=time&start_time=time&item_type=1
+  # GET /rentals/cost?end_time=time&start_time=time&item_types=...
   def cost
-    start_time = Time.zone.parse(params[:start_time]).to_date.to_s
-    end_time = Time.zone.parse(params[:end_time]).to_date.to_s
-    item_type = ItemType.find(params[:item_type]) if params[:item_type]
-    render json: Rental.cost(start_time, end_time, item_type)
+    _params = cost_params
+    required_params = %w(start_time end_time item_types)
+    if (_params.to_h.keys & required_params) == required_params
+      start_time = Time.zone.parse(_params[:start_time]).to_date.to_s
+      end_time = Time.zone.parse(_params[:end_time]).to_date.to_s
+
+      begin
+        cost = _params[:item_types].each_with_object({}) do |it_id, acc|
+          if it = ItemType.find_by(id: it_id)
+            acc[it.name] ||= 0
+            acc[it.name] += it.cost(start_time, end_time)
+          else
+            raise ArgumentError, it_id
+          end
+        end
+      rescue => err
+        render(json: { errors: ["item not found #{err.message}"] }, status: 400) && (return)
+      end
+
+      render json: cost.merge(_total: cost.values.reduce(:+))
+    else
+      render status: 400, json: { errors: [
+        "missing_params: #{(required_params - _params.to_h.keys).inject('') { |acc, part| acc.blank? ? part.to_s : "#{acc}, #{part}" }}"
+      ] }
+    end
   end
 
   # GET /rentals/new
   def new
-    @rental = Rental.new
+    @rental ||= Rental.new # conditionally assign because it could be set already by a method that calls this one
     @start_date = params['start_date'].try(:to_date) || Time.zone.today
     @admin_status = @current_user.has_group? Group.where(name: 'admin')
     set_users_to_assign
+    render :new # this call is required because we directly call the new method on failure of create and other methods
```

Comment:
It is actually not required to write this.
In the create method if you said `render :new` then the create method would render the `new` page, not call this controller action.